llvm/lib/CodeGen/AsmPrinter/DwarfDebug.cpp

```diff
@@ -305,14 +305,17 @@ void Loc::MMI::addFrameIndexExpr(const DIExpression *Expr, int FI) {
 
 static AccelTableKind computeAccelTableKind(unsigned DwarfVersion,
                                             bool GenerateTypeUnits,
+                                            bool HasSplitDwarf,
                                             DebuggerKind Tuning,
                                             const Triple &TT) {
   // Honor an explicit request.
   if (AccelTables != AccelTableKind::Default)
     return AccelTables;
 
-  // Accelerator tables with type units are currently not supported.
-  if (GenerateTypeUnits)
+  // Generating DWARF5 acceleration table.
+  // Currently Split dwarf and non ELF format is not supported.
```

Comment:
Where does the ELF requirement come from?

Reply:
Maybe it's a wrong check? My understanding Mac has their own acceleration tables, not sure they support type units. For Windows I actually don't know.

Reply:
I think we only need to check GenerateTypeUnits !SplitDWARF - the MachO/apple_names case is handled earlier/we don't enable type units in that case at all. And DWARFv5 or greater is the only case that can have debug_names anyway.